google/cloud/storage/internal/storage_stub_factory.cc

```diff
@@ -49,6 +49,10 @@ std::shared_ptr<grpc::Channel> CreateGrpcChannel(
   // to send and/or receive those queries.
   args.SetInt(GRPC_ARG_DNS_ENABLE_SRV_QUERIES, 0);
 
+  if (options.has<AuthorityOption>()) {
```

Comment:
unrelated. send a separate PR?

Reply:
It is related, the code in metrics_exporter_impl does not work without this change, and that is the only motivation for this change.